pkg/agent/agent.go

```diff
@@ -242,14 +237,36 @@ func (a *Agent) bootstrap() error {
 		a.baseSVIDKey = key
 
 		// If we're here, we need to attest/Re-attest
-		regEntryMap, err := a.attest()
+		regEntries, err := a.attest()
 		if err != nil {
 			return err
 		}
-		err = a.FetchSVID(regEntryMap, a.BaseSVID, a.baseSVIDKey)
-		if err != nil {
-			return err
+		serverId := url.URL{
+			Scheme: "spiffe",
+			Host:   a.config.TrustDomain.Host,
+			Path:   path.Join("spiffe", "cp"),
 		}
+		cmgrConfig := &cache.MgrConfig{
+			ServerCerts:    a.serverCerts,
+			ServerSPIFFEID: serverId.String(),
+			ServerAddr:     a.config.ServerAddress.String(),
+
+			BaseSVID:       a.BaseSVID,
+			BaseSVIDKey:    a.baseSVIDKey,
+			BaseRegEntries: regEntries,
+			Logger:         a.config.Log}
+
+		a.CacheMgr, err = cache.NewManager(a.ctx, cmgrConfig)
+
+		go a.CacheMgr.Init()
```

Comment:
Discussed offline implications related to https://github.com/golang/go/issues/20239

Suggest `Init` call to return once the cache manager is running